### ISSUE TITLE
added/removed queries

### ISSUE THREAD

**Issue by ashconnell** [MozillaReality/ecsy#143]
Hi guys, you'll see me lurking around here comparing notes with my own ECS. If I had found ecsy earlier I may have even just used this as they are very similar.

I wanted to share a minor difference in the way I handle queries as it may be useful for ecsy. Queries are set up in much the same fashion except when iterating a query there are 3 arrays: added, active and removed. Ecsy looks to just have "results" which would be kin to my "active" array.

I found this to be hugely beneficial in cutting down the amount of noise in systems, and systems rarely have to do any kind of tracking for cleanups etc. 

Here's an example of how a potential MeshSystem could build/destroy meshes using these queries:

```js
class MeshSystem {
  static queries = {
    default: [MeshComponent]
  }
  
  constructor() {}

  onBeforeStart = () => {
    this.entities = this.queries.default
  }

  onUpdate = () => {
    this.entities.added.forEach(entity => {
      let mesh = this.buildMesh(entity)
      entity.add(Object3D, mesh)
    })
    this.entities.removed.forEach(entity => {
      let mesh = entity.get(Object3D).value
      mesh.dispose()
      entity.remove(Object3D)
    })
  }
}
```

Basically, any entities that match the query for the first time are put in the "added" array and then removed after one frame. The inverse happens with the "removed" array when it no longer matches the query.

I have a similar implementation of `Not` but it's becoming less and less used these days because of these queries.

Any thoughts?

**Comment by ashconnell** [comment]
...I just noticed you already have this under "reactive queries"  lol

**Comment by fernandojsg** [comment]
@ashconnell I feel using `Not` come be more useful when using System State Components similar to what Unity is doing so you don't need to use `added` or `removed`.
Btw regarding `results` I was struggling to find a better name for that, maybe I should change for `active` if that makes more sense... thanks for the suggestion tho ;)
